### ISSUE TITLE
CAN-Festival object dictionary

### ISSUE THREAD

**Issue by framuga** [kcaluwae/dspic33e-bootloader#4]
The "CAN-Festival object dictionary" for the Canopenshell application is not included. Should we modify them with the required entries?
Can this upload dictionaries?
I'm trying to compile using visual C ++ 2017.
Thank you

**Comment by kcaluwae** [comment]
Hi Fran,

I'm not sure if I completely understand what you are trying to do. Do you mind elaborating?

The easiest way to compile the CANOpenShell code (https://github.com/kcaluwae/dspic33e-bootloader/blob/master/host/canfestival/CANOpenShell.c) is by copying it into the canfestival source. Admittedly this is not very clean, but it gets the job done. 
The Python code does essentially the same thing, just without error handling (i.e. it can't be used on a CAN bus with other active CAN devices).

Ken

**Comment by framuga** [comment]
Hello Ken.
I have tried successfully to use the bootloader in a dsPIC33EP512GM604 and the python option of your boot loader.
Now I'm launching it in a dsPIC33EP256MU806, but in the Canfestival version and under visual studio 2017 in windows10.
I have inserted the file "CANOpenShell.c" into the canfestival repository. But in the repository the library objetcs are needed.
In your readme.md you indicate
"The data transfer is based on the CANopen standard, More precisely the object dictionary entry at index 0x1F50, subindex 0x01 is used to program memory pages."
I think this entry is not in the dictionaries that canfestival provides.
When I have already compiled "CANOpenShell.exe" and run in windows the program stops working.

I take this opportunity to tell you again what a great job you have done.
Thank you.

**Comment by framuga** [comment]
Hello Ken.
I have tried successfully to use the bootloader in a dsPIC33EP512GM604 and the python option of your boot loader.
Now I'm launching it in a dsPIC33EP256MU806, but in the Canfestival version and under visual studio 2017 in windows10.
I have inserted the file "CANOpenShell.c" into the canfestival repository. But in the repository the library objetcs are needed.
In your readme.md you indicate
"The data transfer is based on the CANopen standard, More precisely the object dictionary entry at index 0x1F50, subindex 0x01 is used to program memory pages."
I think this entry is not in the dictionaries that canfestival provides.
When I have already compiled "CANOpenShell.exe" and run in windows the program stops working.

I take this opportunity to tell you again what a great job you have done.
Thank you.
